source/game/field/obj/ObjectKinoko.hh

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include "game/field/obj/ObjectCollidable.hh"
+#include "game/field/obj/ObjectKCL.hh"
+
+namespace Field {
+
+enum class KinokoType : u16 {
+    Light = 0,
+    Dark = 1,
+};
+
+/// @brief The class for a mushroom object with normal road properties, for the most part
+class ObjectKinokoNm : public ObjectKCL {
+public:
+    ObjectKinokoNm(const System::MapdataGeoObj &params);
+    ~ObjectKinokoNm() override;
+
+    /// @addr{0x80827A74}
+    [[nodiscard]] const char *getKclName() const override {
+        return m_type == KinokoType::Light ? "kinoko_g" : "kinoko_d_g";
+    }
+
+private:
+    KinokoType m_type;
+};
```

Comment:
Let's move this class (and its subsequent source definitions) down later in the file, below `ObjectKinokoBend`.